src/test/java/com/powsybl/ws/commons/computation/ComputationUtilTest.java

```diff
@@ -0,0 +1,147 @@
+                .add();
+
+        Bus ngen = vl.getBusBreakerView().newBus()
+                .setId("NGEN")
+                .add();
+
+        vl.newLoad()
+            .setId("LD")
+            .setBus(ngen.getId())
+            .setConnectableBus(ngen.getId())
+            .setP0(600.0)
+            .setQ0(200.0)
+            .add();
+
+        return network;
+    }
+
+    public static Network createNodeBreakerNetwork() {
+        Network network = Network.create("network", "test");
+        Substation s = network.newSubstation()
+                .setId("S")
+                .add();
+        VoltageLevel vl = s.newVoltageLevel()
+                .setId("VL1")
+                .setNominalV(400)
+                .setLowVoltageLimit(370.)
+                .setHighVoltageLimit(420.)
+                .setTopologyKind(TopologyKind.NODE_BREAKER)
+                .add();
+        vl.getNodeBreakerView().newBusbarSection()
+                .setId("BBS1")
+                .setNode(0)
+                .add();
+        vl.getNodeBreakerView().newBusbarSection()
+                .setId("BBS2")
+                .setNode(1)
+                .add();
+
+        vl.newLoad()
+            .setId("LD")
+            .setNode(3)
+            .setP0(600.0)
+            .setQ0(200.0)
+            .add();
+
+        vl.getNodeBreakerView().newBreaker()
+            .setId("BR")
+            .setOpen(false)
+            .setNode1(0)
+            .setNode2(3)
+            .add();
+
+        return network;
+    }
+
+    @Test
+    void testViolationLocationIdBusBreaker() {
+        // Setup
+        Network network = createBusBreakerNetwork();
+        BusBreakerViolationLocation busBreakerViolationLocation = mock(BusBreakerViolationLocation.class);
+        when(busBreakerViolationLocation.getType()).thenReturn(ViolationLocation.Type.BUS_BREAKER);
+        when(busBreakerViolationLocation.getBusIds()).thenReturn(List.of("NGEN"));
+        when(busBreakerViolationLocation.getBusView(any())).thenReturn(() -> Stream.empty());
+
+        LimitViolation limitViolation = mock(LimitViolation.class);
+        when(limitViolation.getViolationLocation()).thenReturn(Optional.of(busBreakerViolationLocation));
+        when(limitViolation.getSubjectId()).thenReturn("VLGEN");
+
+        assertEquals("VLGEN", ComputationResultUtils.getViolationLocationId(limitViolation, network));
+
+        VoltageLevel voltageLevel = network.getVoltageLevel("VLGEN");
+        Bus ngen = voltageLevel.getBusView().getBus("VLGEN_0");
+        when(busBreakerViolationLocation.getBusView(any())).thenReturn(() -> Stream.of(ngen));
+        assertEquals("VLGEN_0", ComputationResultUtils.getViolationLocationId(limitViolation, network));
+
+        Bus ngen2 = voltageLevel.getBusBreakerView().newBus()
+                .setId("NGEN2")
+                .add();
+
+        when(busBreakerViolationLocation.getBusView(any())).thenReturn(() -> Stream.of(ngen, ngen2));
```

Comment:
ngen2 is a configured bus and not a calculated bus !

Reply:
done